lib/utils.py

```diff
@@ -57,39 +57,42 @@ def make_request(
     url = BASE_URL + site
 
     attempts = 0
-    while attempts < max_attempts:
-        attempts += 1
-        if method == "POST":
-            response = requests.post(url, headers=headers, json=info)
-        else:
-            response = requests.get(url, headers=headers, params=info)
-
-        if response.status_code == 200:
-            logger.debug("Successfully made request after %d attempts", attempts)
-            return response.json()
-
-        # Request did not succeed
-        response_body = response.content.decode()
-        error_msg = response.reason + " " + str(response.status_code)
-        error = RequestError(error_msg, response_body)
-
-        try:
-            _handle_southwest_error_code(error)
-        except (RequestError, AirportCheckInError) as err:
-            # Stop requesting after one attempt for special codes, as the requests won't succeed
-            error = err
-            break
-
-        if random_sleep:
-            sleep_time = random_sleep_duration(1, 3)
-        else:
-            sleep_time = 0.5
-
-        logger.debug(
-            f"Request error on attempt {attempts}: {error_msg}. Sleeping for {sleep_time:.2f} "
-            "seconds until next attempt"
-        )
-        time.sleep(sleep_time)
+    error = None
+
+    with requests.Session() as session:
+        while attempts < max_attempts:
+            attempts += 1
+            try:
```

Comment:
What’s the reason for moving the try block from line 79 to here. AFAIK, the only function that can throw a `RequestError` or `AurportCheckInError` is `_handle_southwest_error_code`